### DISCUSSION TITLE
Use my setup wifi code is no work

### DISCUSSION THREAD

**Discussion by liulee027** [HomeSpan/HomeSpan#739] [Q&A] [answered]
I tested using honespan set wifi,it is work👍,it can be found in homekit。
But I want to make a network configuration HTML page to connect WIFI
This is the simplified code
I can see the device in the router
But no new devices can be found in homekit
this is part of my code

`#include <WiFi.h>

const char* SSID = "********";
const char* WIfiPasscode = "*********";

void ConnectNet(){
    WiFi.persistent(false);
	WiFi.mode(WIFI_STA);
	WiFi.setAutoReconnect(true);
	WiFi.begin(SSID, WIfiPasscode);
	Serial.println("WiFi connecting...");
	while (!WiFi.isConnected()) {
		delay(100);
		Serial.print(".");
	}
	Serial.print("\n");
	Serial.printf("WiFi connected, IP: %s\n", WiFi.localIP().toString().c_str());
}`

`void setup() {
  // put your setup code here, to run once:
  Serial.begin(115200);
  Serial.println(".");
  Serial.println("Statrt");

  ConnectNet();

  homeSpan.setPairingCode("11111111");
  homeSpan.begin(Category::Lighting,"Switch");  
 
  new SpanAccessory();    
  new Service::AccessoryInformation();  
  new Characteristic::Identify();     
  new HomeSwitch(5);

}`

`void loop() {
  // put your main code here, to run repeatedly:
  button.tick();

  if (!WiFi.isConnected()) {
    ConnectNet();
  }

  homeSpan.poll();   
 
}`

**Comment by HomeSpan** [comment, marked as answer]
I do not understand what you are trying to, but note there is already an HTML network configuration page built into HomeSpan you can use if you want.  See the User Guide page for details.

**Reply by liulee027** [reply]
thank you. i found void setApFunction(void (*func)()) to do it
